tests/e2e/e2e_multiple_services_per_pod_test.go

```diff
@@ -0,0 +1,142 @@
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/openservicemesh/osm/tests/framework"
+)
+
+var _ = OSMDescribe("Test access via multiple services matching the same pod",
+	OSMDescribeInfo{
+		Tier:   2,
+		Bucket: 2,
+	},
+	func() {
+		Context("Multiple services matching same pod", func() {
+			testMultipleServicePerPod()
+		})
+	})
+
+func testMultipleServicePerPod() {
+	{
+		const sourceName = "client"
+		const destName = "server"
+		var ns = []string{sourceName, destName}
+
+		It("Tests traffic to multiple services matching the same pod", func() {
```

Comment:
I would add a blurb here (a few lines) that state how this would work at a glance.
Is this Pod A serving services X and Y at the same time?
Then we'll have `Pod K -> service X -> pod A`
and we'll have `Pod L -> service Y -> pod A`
Is that how this works? Would be very helpful to add these few lines here.